dpp/src/state_transition/validation/validator_state_transition_identity_signature.rs

```diff
@@ -0,0 +1,264 @@
+use anyhow::{anyhow, bail};
+
+use crate::{
+    consensus::signature::SignatureError,
+    identity::{validation::validate_identity_existence, KeyType},
+    prelude::{Identifier, IdentityPublicKey},
+    state_repository::StateRepositoryLike,
+    state_transition::{
+        StateTransition, StateTransitionIdentitySigned, StateTransitionLike, StateTransitionType,
+    },
+    validation::ValidationResult,
+    ProtocolError,
+};
+
+pub async fn validate_state_transition_identity_signature(
+    state_repository: &impl StateRepositoryLike,
+    state_transition: &impl StateTransitionIdentitySigned,
+) -> Result<ValidationResult<()>, ProtocolError> {
+    let mut validation_result = ValidationResult::<()>::default();
+
+    // Owner must exist
+    let result =
+        validate_identity_existence(state_repository, state_transition.get_owner_id()).await?;
+    if !result.is_valid() {
+        return Ok(result.into_result_without_data());
+    }
+
+    let identity = result
+        .data()
+        .ok_or_else(|| anyhow!("the result doesn't contain any Identity"))?;
+    let signature_public_key_id = state_transition.get_signature_public_key_id();
+    let maybe_public_key = identity.get_public_key_by_id(signature_public_key_id);
+
+    let public_key = match maybe_public_key {
+        None => {
+            validation_result.add_error(SignatureError::MissingPublicKeyError {
+                public_key_id: signature_public_key_id,
+            });
+            return Ok(validation_result);
+        }
+        Some(id) => id,
```

Comment:
Maybe rename it to `pk` instead of `id`? :)

Reply:
good point, fixed